### ISSUE TITLE
Microsoft.Extensions.DependencyModel 2.1.1 is missing from nuget

### ISSUE THREAD

**Issue by paulallington** [dotnet/core#1707]
All other libraries I can see (am using) have upgraded to 2.1.1 in nuget, but Microsoft.Extensions.DependencyModel appears to not have happened, even though other libraries have dependencies on it.  Is there a timeline on this?

**Comment by DamianEdwards** [comment]
Which packages are claiming a dependency on `Microsoft.Extensions.DependencyModel` 2.1.1?

**Comment by poteb** [comment]
Could this be the reason, that my ASP.NET Core 2.1 API crashes if I update Microsoft.AspNetCore.All from 2.1.0 to 2.1.1? It doesn't even give me an error, the application never starts up, the debugger just stops. There is nothing in the event log.

**Comment by garethrampton** [comment]
@poteb Have you installed the 2.1.301 SDK/runtime?

It appears the 2.1.1 packages get released on nuget well in advance of the SDK/runtime required to actually use them becomes readily available!

**Comment by kieronlanning** [comment]
@DamianEdwards Is there a plan to release the SDK et al for 2.1.1 anytime soon?

Or maybe pull the 2.1.1 package from NuGet until those bites are released?

**Comment by DamianEdwards** [comment]
Unfortunately the SDK download has got caught up in an issue getting it posted to the download center. We're working to resolve it. We don't have a mechanism for removing packages from NuGet so this is just an unfortunate situation right now.

Please note, for ASP.NET Core projects targeting `netcoreapp2.1` you **don't need to specify a version number for the Microsoft.AspNetCore.App package reference**. This version is now implied based on the SDK version you have installed. If you create a new ASP.NET Core 2.1 project you'll see what I mean:

``` xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>

</Project>
```

This is covered in the migration doc: https://docs.microsoft.com/en-us/aspnet/core/migration/20_21?view=aspnetcore-2.1

You're update process now simply is wait until the SDK is released and then install that. Once that's done, you'll start running on the new version. For packages that aren't included in `Microsoft.AspNetCore.App`, you'll continue to update those the usual way on NuGet, but do so **after** you've installed the new SDK.

**Comment by kieronlanning** [comment]
I guess the issue is that project, like libraries, don't reference that NuGet package. They'll reference the individual packages. And even if they did, because they don't havce the `Microsoft.NET.Sdk.Web` target, they'll still have the reference.

This is especially true of projects that have gone through an upgrade cycle of updating/ upgrading from .NET Core 1.x.

This obviously isn't a show-stopper, but I can imagine that lots of people have wasted a lot of time globally updating then trying to figure out a series of strange build errors.

**Comment by DamianEdwards** [comment]
Understood, and this is exactly why we made this change in 2.1, as folks have been suffering this issue in some form since 1.0. We're thinking about ways we can make it more obvious in a future update (perhaps emit a build warning if the version is present).

**Comment by kieronlanning** [comment]
It looks like https://www.microsoft.com/net/download/ now has the latest bits available.

**Comment by Petermarcu** [comment]
With the release now completely uploaded and available, is this still an issue?

**Comment by kieronlanning** [comment]
I guess that's for @paulallington as the issue owner...?

For me, the release has solved local build issues. Now it's down to build servers and App Services.

**Comment by poteb** [comment]
I can confirm, that the problem is solved after installing SDK 2.1.301.

**Comment by Petermarcu** [comment]
ok, I'm going to close this. We can reopen if issue still persist.
